src/plugins/data/public/antlr/opensearch_sql/code_completion.ts

```diff
@@ -44,32 +43,33 @@ export const getSuggestions = async ({
   selectionEnd,
   position,
   query,
-  connectionService,
-}: QuerySuggestionGetFnArgs): Promise<ISuggestionItem[]> => {
-  const { api } = getUiSettings();
+  openSearchDashboards,
+}: QuerySuggestionGetFnArgs): Promise<QuerySuggestion[]> => {
+  const { api } = openSearchDashboards.uiSettings;
+  const dataSetManager = openSearchDashboards.data.query.dataSet;
+  const selectedDataSet = dataSetManager.getDataSet();
```

Comment:
can you use the `useDatasetManager` hook to get a stateful dataset. See here for reference: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/public/ui/dataset_navigator/dataset_navigator.tsx#L83

Reply:
this is not a react component.